epoll/src/main/scala/fs2/io/epoll/unsafe/libc.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.io.epoll
+package unsafe
+
+import jnr.ffi.{LibraryLoader, Platform, Pointer, Struct, Runtime, LastError, Memory, Union}
+
+private[epoll] object libc {
+
+  object jnr {
+    lazy val globalRuntime = Runtime.getSystemRuntime
+
+    private lazy val libc = LibraryLoader
+      .create(classOf[jnrLibcFFIInterface])
+      .load(Platform.getNativePlatform().getStandardCLibraryName())
+
+    final class EpollData(runtime: Runtime) extends Union(runtime) {
+      final val ptr: PointerField = new Pointer()
+      final val fd: Signed32 = new Signed32()
+      final val u32: Unsigned32 = new Unsigned32()
+      final val u64: Unsigned64 = new Unsigned64()
+    }
+
+    final class EpollEvent(runtime: Runtime) extends Struct(runtime) {
+      final private val _events: Unsigned32 = new Unsigned32()
+      final private val _data: EpollData = inner(new EpollData(runtime))
+
+      @inline
+      final def events(): Int = _events.get().toInt
+      @inline
+      final def fd(): Int = _data.fd.get().toInt
+    }
+
+    object EpollEvent {
+      final val CStructSize: Int = 12
+
+      def apply(events: Int, data: Int, runtime: Runtime): EpollEvent = {
+        val ev = new EpollEvent(runtime)
+        ev._events.set(events.toLong)
+        ev._data.fd.set(data)
+        ev
+      }
+
+      def apply(ptr: Pointer, offset: Long, runtime: Runtime): EpollEvent = {
```

Comment:
I'm confused what this method is doing. Is it copying the event from one native memory to another?